install-panel.sh

```diff
@@ -628,63 +661,43 @@ function firewall_ufw {
   ufw status numbered | sed '/v6/d'
 }
 
-function firewall_firewalld {
+firewall_firewalld() {
   echo -e "\n* Enabling firewall_cmd (firewalld)"
-  echo "* Opening port 22 (SSH), 80 (HTTP) and 443 (HTTPS)"
-
-  if [ "$OS_VER_MAJOR" == "7" ]; then
-    # pointing to /dev/null silences the command output
-    echo "* Installing firewall"
-    yum -y -q update > /dev/null
-    yum -y -q install firewalld > /dev/null
-
-    systemctl --now enable firewalld > /dev/null # Start and enable
-    firewall-cmd --add-service=http --permanent -q # Port 80
-    firewall-cmd --add-service=https --permanent -q # Port 443
-    firewall-cmd --add-service=ssh --permanent -q  # Port 22
-    firewall-cmd --reload -q # Enable firewall
-
-  elif [ "$OS_VER_MAJOR" == "8" ]; then
-    # pointing to /dev/null silences the command output
-    echo "* Installing firewall"
-    dnf -y -q update > /dev/null
-    dnf -y -q install firewalld > /dev/null
-
-    systemctl --now enable firewalld > /dev/null # Start and enable
-    firewall-cmd --add-service=http --permanent -q # Port 80
-    firewall-cmd --add-service=https --permanent -q # Port 443
-    firewall-cmd --add-service=ssh --permanent -q  # Port 22
-    firewall-cmd --reload -q # Enable firewall
+  echo "* Opening port 22 (SSH), 80 (HTTP) and (optionaly) 443 (HTTPS)"
 
-  else
-    print_error "Unsupported OS"
-    exit 1
-  fi
+  # Install
+  [ "$OS_VER_MAJOR" == "7" ] && yum -y -q install firewalld > /dev/null
+  [ "$OS_VER_MAJOR" == "8" ] && dnf -y -q install firewalld > /dev/null
+
+  # Enable
+  systemctl --now enable firewalld > /dev/null # Enable and start
+
+  # Configure
+  firewall-cmd --add-service=http --permanent -q # Port 80
+  firewall-cmd --add-service=https --permanent -q # Port 443
+  firewall-cmd --add-service=ssh --permanent -q  # Port 22
+  firewall-cmd --reload -q # Enable firewall
 
   echo "* Firewall-cmd installed"
   print_brake 70
 }
 
-function letsencrypt {
+letsencrypt() {
   FAILED=false
 
   # Install certbot
-  if [ "$OS" == "debian" ] || [ "$OS" == "ubuntu" ]; then
-    apt-get install certbot -y
-  elif [ "$OS" == "centos" ]; then
-    [ "$OS_VER_MAJOR" == "7" ] && yum install certbot
-    [ "$OS_VER_MAJOR" == "8" ] && dnf install certbot
-  else
-    # exit
-    print_error "OS not supported."
-    exit 1
-  fi
-
-  # Stop nginx
-  systemctl stop nginx
+  case "$OS" in
+    debian | ubuntu)
+      apt-get -y install certbot python3-certbot-nginx
```

Comment:
Is the `python3-certbot-nginx` required? Can't recall having to install it on Debian 10.

Reply:
```bash
root@localhost:~# apt install certbot
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  python3-acme python3-certbot python3-configargparse python3-josepy python3-mock
  python3-openssl python3-parsedatetime python3-pbr python3-requests-toolbelt
  python3-rfc3339 python3-tz python3-zope.component python3-zope.event
  python3-zope.hookable python3-zope.interface
Suggested packages:
  python3-certbot-apache python3-certbot-nginx python-certbot-doc python-acme-doc
  python-mock-doc python-openssl-doc python3-openssl-dbg
Recommended packages:
  python3-pyicu
The following NEW packages will be installed:
  certbot python3-acme python3-certbot python3-configargparse python3-josepy
  python3-mock python3-openssl python3-parsedatetime python3-pbr
  python3-requests-toolbelt python3-rfc3339 python3-tz python3-zope.component
  python3-zope.event python3-zope.hookable python3-zope.interface
```
This is debian stretch (shouldn't be different from buster). It suggests installing `python3-certbot-nginx `, but doesn't install it. It is needed for `certbot --nginx`